lib/auth/auth.js

```diff
@@ -31,7 +31,7 @@ auth.check = (request, log, awsService, data) => {
             return authV2.headerAuthCheck.check(request, log, data);
         } else if (authHeader.startsWith('AWS4')) {
             log.trace('authenticating request with Auth V4 using headers');
-            return authV4.headerAuthCheck.check(request, log, awsService);
+            return authV4.headerAuthCheck.check(request, log, data, awsService);
```

Comment:
Will you please add a jsdoc for auth.check above?  It will make it clearer what the difference is between request and data.